healthcheck.go

```diff
@@ -71,21 +72,21 @@ func (r *ReplicaSet) Check() error {
 	case err := <-errChan:
 		if err != nil {
 			r.Stats.BumpSum("healthcheck.failed", 1)
-			r.Log.Errorf("Failed healtcheck due to %s", err)
+			r.Log.Errorf("Failed healthcheck due to %s", err)
 		} else {
 			r.Stats.BumpSum("healthcheck.connected", 1)
 		}
 		return err
 	case <-time.After(TIMEOUT):
 		r.Stats.BumpSum("healthcheck.failed", 1)
-		r.Log.Errorf("Failed healtcheck due to timeout %s", TIMEOUT)
+		r.Log.Errorf("Failed healthcheck due to timeout %s", TIMEOUT)
 		return errors.New("Failed due to timeout")
 	}
 }
 
-func (r *ReplicaSet) RestartIfFailed() {
-	r.Log.Error("Restarting replica set due to consecutive failed healthchecks ")
-	go r.Restart()
+func (r *ReplicaSet) HandleFailure() {
+	r.Log.Error("Crashing dvara due to consecutive failed healthchecks")
```

Comment:
Great :)
Could you add a metric for this? 
Ideally monit would do this for us, but we don't have any monit metrics.